### ISSUE TITLE
After updating from Chef-12.21 to chef-15.2.20 , deployments are failing randmly

### ISSUE THREAD

**Issue by infa-thothadv** [chef/chef#9124]
## Description
After upgrading to chef 15.2.20, we are randomly hitting this issue while we are running the chef recipe. We are using chef-solo 

wrong number of arguments (given 0, expected 1+)

## Chef Version
chef 15.2.20

## Platform Version
centos 7.4

## Replication Case

``` this issue is not easily reproducible. We hit it once every 5 times. But due to this our deployments are randomly failing```

## Client Output
```
Starting Chef Infra Client, version 15.2.20^[[0m
[2019-11-07T23:35:59+00:00] WARN: Plugin Network: unable to detect ipaddress
[2019-11-07T23:36:00+00:00] WARN: Run List override has been provided.
[2019-11-07T23:36:00+00:00] WARN: Original Run List: []
[2019-11-07T23:36:00+00:00] WARN: Overridden Run List: [recipe[cloudUI::db]]
resolving cookbooks for run list: ["cloudUI::db"]^[[0m
Synchronizing Cookbooks:^[[0m
  - cloudUI (0.1.76)^[[0m
  - entrypoint (0.1.59)^[[0m
  - tomcat (0.1.108)^[[0m
  - javainstall (0.1.61)^[[0m
  - deploy-services (0.1.55)^[[0m
  - database (0.2.65)^[[0m
  - infa-properties (0.1.128)^[[0m
  - consul-config (0.0.82)^[[0m
  ^[[0m

  ================================================================================^[[0m
  [[31mError Syncing Cookbooks:[[0m
  ================================================================================^[[0m
[[0m  Unexpected Error:[[0m
  -----------------^[[0m
  ArgumentError: wrong number of arguments (given 0, expected 1+)^[[0m
[[0m  System Info:[[0m
  ------------^[[0m
  chef_version=15.2.20
^[[0m  platform=centos
^[[0m  platform_version=7.6.1810
^[[0m  ruby=ruby 2.6.3p62 (2019-04-16 revision 67580) [x86_64-linux]
^[[0m  program_name=/opt/chef/bin/chef-client
[[0m  executable=/opt/chef/bin/chef-client[[0m

^[[0m  ^[[0m
  Running handlers:^[[0m
[2019-11-07T23:36:01+00:00] ERROR: Running exception handlers
  Running handlers complete
^[[0m[2019-11-07T23:36:01+00:00] ERROR: Exception handlers complete
  Chef Infra Client failed. 0 resources updated in 01 seconds^[[0m
  - sensu-config (0.0.62)^[[0m
[2019-11-07T23:36:01+00:00] FATAL: Stacktrace dumped to /opt/entrypoint/local-mode-cache/cache/chef-stacktrace.out
[2019-11-07T23:36:01+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2019-11-07T23:36:01+00:00] FATAL: ArgumentError: wrong number of arguments (given 0, expected 1+)
```

**Comment by lamont-granquist** [comment]
There's not enough information in this to debug, all the useful details of the error message and the stack trace are omitted

**Comment by infa-thothadv** [comment]
This got fixed after we upgraded chef version to 15.4.45
